### ISSUE TITLE
problem with '/' char in filename

### ISSUE THREAD

**Issue by MisterXYZ83** [megous/megatools#248]
There is an issue downloading files/directory with the slash '/' char in nodename; for the moment i solved modifing the mega.c source code below line 1836 adding a simple walkaround (swap the dangerous '/' with a safer '-' in node_name string).

I hope for a more stable solution! thanks

**Comment by megous** [comment]
If mega node contains / or other invalid or "dangerous" characters on windows, it will be ignored completely. I think it's a good idea to ignore these nodes, because they seem obviously malicious and don't have any valid use.

Renaming them may introduce name conflicts with other nodes.

**Comment by MisterXYZ83** [comment]
yes you're right, buf i needed to download a directory on mega called
something like 'games 1/2' ...i am sure it is safe for this archive.

Il 02 gen 2017 02:45, "megous" <notifications@github.com> ha scritto:

**Comment by nyuszika7h** [comment]
I don't know why it is valid in MEGA, but I would suggest sanitizing invalid characters on the filesystem and replacing them with `_`. Though that means either you would need to be stricter about filenames on Linux too, or potentially cause issues when someone transfers their files between Windows and Linux. Maybe add a switch like `--force-restrict-filenames`.

**Comment by megous** [comment]
I think names are sanitized already. It's that invalid characters are removed instead of replaced with _ at the moment.

**Comment by nyuszika7h** [comment]
Maybe I'm on an old version then...

**Comment by megous** [comment]
Ah, I see. The node gets ignored completely:

https://github.com/megous/megatools/blob/master/lib/mega.c#L1904
https://github.com/megous/megatools/blob/master/lib/mega.c#L3636

**Comment by 64617** [comment]
The bug was just fixed (although I figure using a safer character like '_' might have been a better option)
